scripts/upgrade

```diff
@@ -14,28 +12,178 @@ source /usr/share/yunohost/helpers
+ynh_add_systemd_config
+
+#=================================================
+# COPY CONFIG FILE
+#=================================================
+
+if [ -e /home/yunohost.app/$app ]
+then
+echo "Nothing to do..."
+else
+mkdir -p /home/yunohost.app/$app/.minetest/
+mkdir /home/yunohost.app/$app/.minetest/worlds/
+fi
+
+#=================================================
+# SPECIFIC UPGRADE
+#=================================================
+# Compile Minetest
+pushd /opt/yunohost/$app
+# Download Minetest Game
+ynh_setup_source --dest_dir=/opt/yunohost/$app/games/minetest_game --source_id=minetest_game
+
+cmake . -DRUN_IN_PLACE=TRUE -DENABLE_LUAJIT=TRUE -DBUILD_CLIENT=FALSE -DBUILD_SERVER=TRUE
+
+make -j$(nproc)
+
+popd
+
+
+#=================================================
+# MODIFY A CONFIG FILE
+#=================================================
+
+ynh_backup_if_checksum_is_different --file="$home_path/.minetest/minetest.conf"
+
+cp ../conf/minetest.conf $home/.minetest/minetest.conf
+
+### `ynh_replace_string` is used to replace a string in a file.
+### (It's compatible with sed regular expressions syntax)
+
+ynh_replace_string --match_string="__APP__" --replace_string="$app" --target_file="$home_path/.minetest/minetest.conf"
+ynh_replace_string --match_string="__PORT__" --replace_string="$port" --target_file="$home_path/.minetest/minetest.conf"
+ynh_replace_string --match_string="__SERVERNAME__" --replace_string="$servername" --target_file="$home_path/.minetest/minetest.conf"
+ynh_replace_string --match_string="__DOMAIN__" --replace_string="$domain" --target_file="$home_path/.minetest/minetest.conf"
+ynh_replace_string --match_string="__ANNOUNCE__" --replace_string="$is_public" --target_file="$home_path/.minetest/minetest.conf"
+ynh_replace_string --match_string="__PVP__" --replace_string="$pvp" --target_file="$home_path/.minetest/minetest.conf"
+ynh_replace_string --match_string="__CREATIVE__" --replace_string="$creative" --target_file="$home_path/.minetest/minetest.conf"
+ynh_replace_string --match_string="__DAMAGE__" --replace_string="$damage" --target_file="$home_path/.minetest/minetest.conf"
+
+ynh_store_file_checksum --file="$home_path/.minetest/minetest.conf"
+
+#=================================================
+# SETUP LOGROTATE
+#=================================================
+ynh_print_info --message="Upgrading logrotate configuration..."
+
+# Use logrotate to manage app-specific logfile(s)
+ynh_use_logrotate --non-append
+
+
+
+#=================================================
+# GENERIC FINALIZATION
+#=================================================
+# SECURE FILES AND DIRECTORIES
+#=================================================
+
+# Set permissions on app files
+chown -R root: $final_path
+chown -R $app: $home_path
+
+#=================================================
+# RELOAD MINETEST
+#=================================================
+ynh_print_info --message="Restarting Minetest..."
+
+systemctl restart $app
```

Comment:
```suggestion
ynh_systemd_action --action=restart --line_match="Server for gameid="$app" listening on :::30000." --log_path="/var/log/$app/minetest.log"
```

Reply:
oups!! sorry!

We have to escape the " char:

`ynh_systemd_action --action=restart --line_match="Server for gameid=\"$app\" listening on :::30000." --log_path="/var/log/$app/minetest.log"`

is much better! (To be use everywhere)

Reply:
Bump